hardware/Netatmo.cpp

```diff
@@ -488,14 +526,146 @@ uint64_t CNetatmo::convert_mac(std::string mac)
 }
 
 
+uint64_t getValueFromString(const std::string& value)
+{
+	uint64_t result;
+	std::istringstream iss( value );
+	iss >> result;
+	return result;
+}
+
+std::string getStringFromValue(const uint64_t value)
+{
+	std::stringstream stream;
+	stream << value;
+	std::string string (stream.str());
+	return string;
+}
+
+/// <summery>
+/// Merge old and new devices, bu deleting duplicate log records and
+/// copy all other log records from the new to the old device before
+/// updating the old device to the new devices and finaly remove the new device.
+/// Note: the old device will become the new device.
+/// </summary>
+bool CNetatmo::MergeDevices (const uint64_t ipOldDeviceId, const uint64_t ipNewDeviceId)
+{
+//	Log(LOG_STATUS, "MergeDevices: Merge old device %" PRIu64 " with new device %" PRIu64 ".", ipOldDeviceId, ipNewDeviceId);
+
+	std::string sOldDeviceId = getStringFromValue(ipOldDeviceId);
+	std::string sNewDeviceId = getStringFromValue(ipNewDeviceId);
+
+	Log(LOG_STATUS, "MergeDevices: Merge old device %s with new device %s", sOldDeviceId.c_str(), sNewDeviceId.c_str());
+
+	auto result = m_sql.safe_query
+	(
+		"SELECT id FROM DeviceStatus  WHERE Id='%q' OR Id = '%q';",
+		sOldDeviceId.c_str(),
+		sNewDeviceId.c_str()
+	);
+	if (result.empty())
```

Comment:
Investigating....

Reply:
These lines were meant as general check on all kinds of (external caused) errors (like e.g. a database lock) and a sanity check. I was learned to test always on the unexpected...

I'll remove them, because there is no need for this query anymore.

Reply:
Done (removed)